agent/src/test/java/com/microsoft/applicationinsights/agent/internal/config/XmlAgentConfigurationBuilderTest.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent.internal.config;
+
+import com.microsoft.applicationinsights.agent.internal.agent.ClassInstrumentationData;
+import org.apache.commons.io.FileUtils;
+import org.junit.Test;
+
+import java.io.*;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+public final class XmlAgentConfigurationBuilderTest {
+    private final static String TEMP_TEST_FOLDER = "AgentTests";
+    private final static String TEMP_CONF_FILE = "AI-Agent.xml";
+
+    private interface Printer {
+        void print(PrintWriter writer);
+    }
+
+    private static class MalformedXmlTestPrinter implements Printer {
+        @Override
+        public void print(PrintWriter writer) {
+            writer.println("<BuiltIn>");
+        }
+    }
+
+    private static class BuiltInTestPrinter implements Printer {
+        @Override
+        public void print(PrintWriter writer) {
+            writer.println("<BuiltIn>");
+
+            writer.println("<HIBERNATE enabled=\"false\"/>");
+            writer.println("<HTTP enabled=\"true\"/>");
+
+            writer.println("</BuiltIn>");
+        }
+    }
+
+    private static class ForbiddenSectionTestPrinter implements Printer {
+        @Override
+        public void print(PrintWriter writer) {
+            writer.println("<ForbiddenPrefixes>");
```

Comment:
Please take this XML content out into an xml file and load it as string from there (instead of building its string like this).